docs/sql-ref-syntax-qry-select-join.md

```diff
@@ -18,5 +18,222 @@ license: |
+      <code>
+      ON boolean_expression | USING ( column_name [ , column_name ... ] )
+      </code> <br><br>
+      <code>boolean_expression</code><br>
+      Specifies an expression with a return type of boolean.
+
+  </dd>
+</dl>
+
+### Join Types
+
+- <b>Inner Join </b><br>
+The inner join is the default join in Spark. It selects rows that have matching values in both relations.<br><br>
+  <b>Syntax:</b><br>
+    <code>
+    relation [ INNER ] JOIN relation [ join_criteria ]
+    </code>
+
+- <b>Left Join </b><br>
+A left join returns all values from the left relation and the matched values from the right relation, or appends NULL if there is no match. It is also referred to as a left outer join.<br><br>
+  <b>Syntax:</b><br>
+    <code>
+    relation LEFT [ OUTER ] JOIN relation [ join_criteria ]
+    </code>
+
+- <b>Right Join </b><br>
+A right join returns all values from the right relation and the matched values from the left relation, or appends NULL if there is no match. It is also referred to as a right outer join.<br><br>
+  <b>Syntax:</b><br>
+    <code>
+    relation RIGHT [ OUTER ] JOIN relation [ join_criteria ]
+    </code>
+
+- <b>Full Join </b><br>
+A full join returns all values from both relations, appending NULL values on the side that does not have a match. It is also referred to as a full outer join.<br><br>
+  <b>Syntax:</b><br>
+    <code>
+    relation FULL [ OUTER ] JOIN relation [ join_criteria ]
+    </code>
+
+- <b>Cross Join </b><br>
+A cross join returns the Cartesian product of two relations.<br><br>
+  <b>Syntax:</b><br>
+    <code>
+    relation CROSS JOIN relation [ join_criteria ]
+    </code>
+
+- <b>Semi Join </b><br>
+A semi join returns values from the left side of the relation that has a match with the right. It is also referred to as a left semi join.<br><br>
+  <b>Syntax:</b><br>
+    <code>
+    relation [ LEFT ] SEMI JOIN relation [ join_criteria ]
+    </code>
+
+- <b>Anti Join </b><br>
+An anti join returns values from the left relation that has no match with the right. It is also referred to as a left anti join.<br><br>
+  <b>Syntax:</b><br>
+    <code>
+    relation [ LEFT ] ANTI JOIN relation [ join_criteria ]
+    </code>
+
+### Examples
+{% highlight sql %}
+-- Use employee and department tables to demonstrate different type of joins.
+SELECT * FROM employee;
+
+  +---+-----+-------+
+  |id |name |deptno |
+  +---+-----+-------+
+  |105|Chloe|5      |
+  |103|Paul |3      |
+  |101|John |1      |
+  |102|Lisa |2      |
+  |104|Evan |4      |
+  |106|Amy  |6      |
+  +---+-----+-------+
+
+SELECT * FROM department;
+  +-------+-----------+
+  |deptno |deptname   |
+  +-------+-----------+
+  |3      |Engineering|
+  |2      |Sales      |
+  |1      |Marketing  |
+  +-------+-----------+
+
+-- Use employee and department tables to demonstrate inner join.
+SELECT id, name, employee.deptno, deptname
+  FROM employee INNER JOIN department ON employee.deptno = department.deptno;
```

Comment:
nit: 2 space or 4 spaces in the head? I found 3 spaces... e.g., https://github.com/apache/spark/blame/master/docs/sql-ref-syntax-qry-select-groupby.md#L130

Reply:
I prefer either 2 or 4 spaces. 
I checked a few SQL Reference:
DB2, Snowflake 2 spaces
SQL Server, PostgreSQL 4 spaces
Oracle 3 spaces

Reply:
Ah, I see. Anything looks fine to me, but we need consistency in our SQL docs.